dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -2492,6 +2548,13 @@ Sedp::signal_liveliness_secure(DDS::LivelinessQosPolicyKind kind)
 }
 #endif
 
+ICE::Endpoint* Sedp::get_ice_endpoint() {
+  DCPS::RtpsUdpInst_rch rtps_inst =
+    DCPS::static_rchandle_cast<DCPS::RtpsUdpInst>(transport_inst_);
```

Comment:
Is this cast still needed?  get_ice_endpoint is defined in `TransportImpl`.

Reply:
The impl() method is protected in the base but public in the derived class.

Reply:
Instead of depending on the general-purpose `impl()`, how about adding a method to `TransportInst` itself to support this use?